pallets/rmrk-core/src/lib.rs

```diff
@@ -403,10 +422,32 @@ pub mod pallet {
 				new_owner.clone(),
 				max_recursions,
 			)?;
+			// Check if root owner is a nft & call pallet_uniques::do_transfer w/ actual account owner
+			match Pallet::<T>::decode_nft_account_id::<T::AccountId>(new_owner_account.clone()) {
+				None => {
+					pallet_uniques::Pallet::<T>::do_transfer(
+						collection_id,
+						nft_id,
+						new_owner_account.clone(),
+						|class_details, details|
+							Ok(())
+					)?;
+				},
+				Some((cid, nid)) => {
+					let (dest, _) = Pallet::<T>::lookup_root_owner(cid, nid)?;
+					pallet_uniques::Pallet::<T>::do_transfer(
+						collection_id,
+						nft_id,
+						dest.clone(),
+						|class_details, details|
+							Ok(())
+					)?;
+				}
+			}
```

Comment:
After the transfer, we need to also maintain the `Children` storage. I suggest to add a unit test to cover this property